cmd/nerdctl/multi_platform_linux_test.go

```diff
@@ -159,6 +166,10 @@ RUN uname -m > /usr/share/nginx/html/index.html
 		"http://127.0.0.1:8081": "aarch64",
 		"http://127.0.0.1:8082": "armv7l",
 	}
+	// using aarch64 instead of armv7l because the arm64 system is compatible with armv7.
```

Comment:
Hey @yankay I do not understand why we should do that / what is it bringing us.

Reply:
HI @apostasie 

Ref to https://github.com/containerd/containerd/issues/10466. The `ctr` has the same behavior as `nerdctl`.
When we run armv7 on x86, the `binfmt` uses qemu to emulate. So the `uname -m` commands show `armv7` .
But when we run armv7 on aarch64, it does not need to emulate the Linux runs the binary directly. So the `uname -m` commands show `aarch64` .

Ref to: https://github.com/docker/buildx/issues/198

Reply:
My host is on arm64 (with qemu installed, as we do on the CI):
```
$ uname -m
aarch64
$ nerdctl run --rm --platform arm alpine uname -m
armv7l
```

Reply:
Tonis comment (in https://github.com/docker/buildx/issues/198#issuecomment-561252903 ) makes it clear that this is without qemu.

But here *we do* enable qemu - which is why I am confused at how this could work (especially since it does not, locally)

Reply:
MY test environment is:
```
uname -a
Linux iZj6c145pjnolrzzpcuvx2Z 5.15.0-76-generic #83-Ubuntu SMP Thu Jun 15 19:21:56 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```
Does your host MacBook arm64 ?

Reply:
Clearly the call to binfmt on the CI did not install the arm emulator, while it did on my machine. As to the why, my money is on the fact that the M1 may not have the magic native execution fallback thing.

Anyhow:
a. we should pin toniis image instead of using latest
b. looking at the source of it, it has clearly grown in complexity and trying too hard to do "magic" IMHO - `--install all` clearly no longer means `--install all`, it is now "install what it thinks you need". I would suggest we stop using it entirely or at least we stop doing `--install-all` and we instead explicitly install the architectures we need

Reply:
Thanks @apostasie . It works. 👍